optuna/study/study.py

```diff
@@ -543,9 +604,14 @@ def df(x):
                 is :class:`~optuna.trial.TrialState.FAIL` or
                 :class:`~optuna.trial.TrialState.PRUNED`.
             state:
-                State to be reported. Must be :class:`~optuna.trial.TrialState.COMPLETE`,
+                State to be reported. Must be :obj:`None`,
+                :class:`~optuna.trial.TrialState.COMPLETE`,
                 :class:`~optuna.trial.TrialState.FAIL` or
                 :class:`~optuna.trial.TrialState.PRUNED`.
+                If ``state`` is :obj:`None`,
+                it will be updated to :class:`~optuna.trial.TrialState.COMPLETE`
+                or :class:`~optuna.trial.TrialState.FAIL` depending on whether
+                a validation for ``values`` reported succeed or not.
```

Comment:
Looks but, just a small nit.
```suggestion
                validation for ``values`` reported succeed or not.
```